shared/src/commonMain/kotlin/com/projectnewm/BaseUseCase.kt

```diff
@@ -0,0 +1,9 @@
+package com.projectnewm
+
+import com.projectnewm.util.CommonFlow
+import com.projectnewm.util.DataState
+
+interface BaseUseCase<T> {
+
+    fun execute(): CommonFlow<DataState<T>>
```

Comment:
Note: generic interfaces don't map to Swift, so this `T` gets converted to Swift's `AnyObject`.

Reply:
Aw bummer! Well, we don't really need it. I'll remove the Base class